interp/api.go

```diff
@@ -71,10 +71,12 @@ type Runner struct {
 	// arguments. It may be nil.
 	callHandler CallHandlerFunc
 
-	// execHandler is a function responsible for executing programs. It must be non-nil.
+	// execHandler is responsible for executing programs. It must not be nil.
 	execHandler ExecHandlerFunc
 
-	// openHandler is a function responsible for opening files. It must be non-nil.
+	execMiddlewares []func(ExecHandlerFunc) ExecHandlerFunc
```

Comment:
You could potentially move this type to something like this, but this is only a suggestion 🙂 

```go
type ExecMiddlewareFunc func(ExecHandlerFunc) ExecHandlerFunc
```

```go
execMiddlewares []ExecMiddlewareFunc
```

Reply:
I thought about a type, but note that we don't want a named type - we don't need methods and I don't want to add type conversions into the mix.

A type alias wouldn't have that problem, but it doesn't really gain us much. The API would go from

```
func ExecHandlers(middlewares ...func(next ExecHandlerFunc) ExecHandlerFunc) RunnerOption
```

to

```
type ExecMiddleware = func(ExecHandlerFunc) ExecHandlerFunc

func ExecHandlers(middlewares ...ExecMiddleware) RunnerOption
```

so in my opinion this is only worthwhile if the `func(T) T` type appears in multiple places. It might in the future, but it doesn't now, so I don't think it's worth it :)

Reply:
Adding godoc.